### PR TITLE
Make function write_to_file use open(..) as a context manager

### PR DESCRIPTION
**Pull request by hartwork**
.. and handle `OSError` exceptions after opening the file.

### CODE REVIEW

File: git_big_picture/_main.py [hunk 1]
```diff
@@ -498,13 +498,12 @@ def write_to_file(output_file, dot_output):
 
     """
     try:
-        f = open(output_file, 'wb+')
+        with open(output_file, 'wb+') as f:
+            f.write(dot_output)
+            f.flush()
```
Comment (esc) [inline git_big_picture/_main.py:503]: Quick question, are the flush and fsync needed, even if we are doing this inside a context manager?

Comment (hartwork) [reply git_big_picture/_main.py:503]: It *may* not be needed, but:
- I can only "guarantee" a call to `open` as a context manager calls `.close`, nothing more
- We may have had some good reason to add it back then (needs a closer look), and before being sure, we may produce a regression.

@esc so unless you can prove these calls as no longer needed, my vote is for not dropping them due to risk and low current pain.

PS: From https://docs.python.org/3/library/io.html#io.TextIOWrapper:
> [IOBase](https://docs.python.org/3/library/io.html#io.IOBase) is also a context manager and therefore supports the [with](https://docs.python.org/3/reference/compound_stmts.html#with) statement. In this example, file is closed after the with statement’s suite is finished—even if an exception occurs:

Comment (esc) [reply git_big_picture/_main.py:503]: OK, it's not worth the effort, let's just keep them and be safe.

**Review by esc — APPROVED** [review]
LGTM

**Pull request merged**